### ISSUE TITLE
Batch Prediction on Multiple Images

### ISSUE THREAD

**Issue by dgzct11** [amirhosseinh77/JetsonYolo#7]
Hello amirhosseinh77,

I am attempting to process video from 2 video USB cameras at the same time. I have been using numpy.stack to combine the images into one frame, but this decreases the accuracy due to the lower final resolution. 

For the original yolov5 repository, I can batch the two frames together to increase the fps and accuracy, and I was wondering if I could do that with Jetson Yolo?

Thank you

**Comment by arbit3rr** [comment]
Hi there, yes, it is possible but it needs some modifications.
I will try to explain the code here and I suggest you try it on your computer and take it to your board then.
You need to make changes to elements/yolo.py in the detect method:
repeat :

        height, width = main_img.shape[:2]
        new_height = int((((self.input_width/width)*height)//32)*32)
        img = cv2.resize(main_img, (self.input_width,new_height))
        img = cv2.cvtColor(img,cv2.COLOR_BGR2RGB)
        img = np.moveaxis(img,-1,0)
        img = torch.from_numpy(img).to(device)
        img = img.float()/255.0  # 0 - 255 to 0.0 - 1.0
        if img.ndimension() == 3:
            img = img.unsqueeze(0)

for any image you have and stack them to gether then pass it to the model:
        pred = self.yolo_model(img, augment=False)[0]
        pred = non_max_suppression(pred, conf_thres=0.25, iou_thres=0.45, classes=None)

you also have to repeat the next part for both images (not only pred[0]) :

        if pred[0] is not None and len(pred):
            for p in pred[0]:
                score = np.round(p[4].cpu().detach().numpy(),2)
                label = self.classes[int(p[5])]
                xmin = int(p[0] * main_img.shape[1] /self.input_width)
                ymin = int(p[1] * main_img.shape[0] /new_height)
                xmax = int(p[2] * main_img.shape[1] /self.input_width)
                ymax = int(p[3] * main_img.shape[0] /new_height)

                item = {'label': label,
                        'bbox' : [(xmin,ymin),(xmax,ymax)],
                        'score': score
                        }
